src/rv/comm/rcssserver/GameState.java

```diff
@@ -48,6 +48,34 @@
         void gsServerMessageProcessed(GameState gs);
     }
 
+    public enum FoulType {
+        CROWDING(0, "crowding"), TOUCHING(1, "touching"), ILLEGALDEFENCE(2, "illegal defence"),
+        ILLEGALATTACK(3, "illegal attack"), INCAPABLE(4, "incapable"),
+        KICKOFF(5, "illegal kickoff"), CHARGING(6, "charging");
+
+        private int    idx;
```

Comment:
I didn't originally write this code, but I think the point is that `idx` (index) is the index of the foul types coming from the server, and need to match with the server index types interpreted in `GameState.FoulType.values()[Integer.parseInt(atoms[2])]`.  'str' is the string written in the foul list to represent that type of foul.